src/DIRAC/TransformationSystem/Client/TaskManager.py

```diff
@@ -68,7 +68,7 @@ def updateDBAfterTaskSubmission(self, taskDict):
         for taskID, task in taskDict.items():
             transID = task["TransformationID"]
             if task["Success"]:
-                res = self.transClient.setTaskStatusAndWmsID(transID, taskID, "Submitted", str(task["ExternalID"]))
+                res = self.transClient.setTaskStatusAndWmsID(transID, int(taskID), "Submitted", str(task["ExternalID"]))
```

Comment:
You are correct that it very likely comes from `JEncode`, but I do not think this is the correct fix. I assume that `taskID` should already be an `int`.
The only place I can see `updateDBAfterTaskSubmission` is 
https://github.com/DIRACGrid/DIRAC/blob/1a6fc08cf01e1fa2431eef5da5209efd848ffc3e/src/DIRAC/TransformationSystem/Agent/TaskManagerAgentBase.py#L663

Looking up the chain, my unntested guess is that we should implement `getTasksToSubmit` in the client with the proper conversion. See for example:

https://github.com/DIRACGrid/DIRAC/blob/1a6fc08cf01e1fa2431eef5da5209efd848ffc3e/src/DIRAC/DataManagementSystem/Service/FTS3ManagerHandler.py#L156-L164

Reply:
You could test your assumption by adding the conversion here, I guess

https://github.com/DIRACGrid/DIRAC/blob/1a6fc08cf01e1fa2431eef5da5209efd848ffc3e/src/DIRAC/TransformationSystem/Agent/TaskManagerAgentBase.py#L607

Reply:
I don't have an issue trying the proper fix as a hotfix, but I do I trigger the problem on the certif instance ?

Reply:
The test from here https://trello.com/c/Rk7HHlxy/6-submit-replication-transformation